### ISSUE TITLE
Content-Type header handling

### ISSUE THREAD

**Issue by hugeval** [zendframework/zend-mail#107]
Hi

There is a case when emails have Content-Type header, defined with keyword "ContentType", for example:
```
ContentType: text/html; charset="UTF-8"
```

In this case error appears:
```
Invalid header line for Content-Type string
```

[Looking header fieldName normalization](https://github.com/zendframework/zend-mail/blob/master/src/Headers.php#L524), I see that `Content-Type`, `Content_Type`, `Content Type`, `ContentType` will be normalized to `contenttype`. So `contenttype` will be stored in `Zend\Mail\Headers::$headersKeys` and appropriate header in `Zend\Mail\Headers::$headers`.
[Than on header loading](https://github.com/zendframework/zend-mail/blob/master/src/Headers.php#L494) pluginClassLoader will load `Zend\Mail\Header\ContentType` and method `Zend\Mail\Header\ContentType::fromString()` is called.
But (there is condition)[https://github.com/zendframework/zend-mail/blob/master/src/Header/ContentType.php#L40] that do not accept 'contenttype`  value. 

It looks like inconsistent behavior. I understand that [rfc2045](https://tools.ietf.org/html/rfc2045#section-5) accepts `Content-Type` keyword only, but in real cases other keywords can appear.

**Comment by adamlundrigan** [comment]
Good catch :+1: Definitely looks to me like a strictness mismatch between the two classes:  `ContentType::fromString` does not normalize the keyword while the Headers bag does.  I'm not familiar enough with Zend\Mail to know which is the intended behaviour, but it should definitely be consistent.  Tecnically, only `Content-Type` is valid and the others should be ignored, but our world is not that perfect :stuck_out_tongue: 

I spot-checked a few other Headers classes for headers with hyphenated names and those also do not do normalization and so would have the same problem.

**Comment by Freeaqingme** [comment]
I'm not sure we should support the contentType header. I just searched through a corpus of about a million emails sent over the past 5 days. And found only 5 emails containing the 'contentType' header. All from the same server and domain, and all mails with the contentType header also had a Content-Type header (with an identical value).

I could search in a much larger corpus as well, but this server is handling email for about a 100K domains, so I'd say it's fairly representative.

**Comment by hugeval** [comment]
@Freeaqingme , I'm not sure, but it looks like, mails sent using Microsoft Outlook have both `Content-Type` and `ContentType` headers in each email.

**Comment by adamlundrigan** [comment]
@Freeaqingme how about other headers this class normalizes the field names of?  Could we drop [that normalization](https://github.com/zendframework/zend-mail/blob/master/src/Headers.php#L524) for 3.0?  Right now it's just shifting the problem from the header bag into the individual header classes (which reject the invalid field names).

**Comment by fasterforward** [comment]
We are also experiencing this issue. Mail which is sent by Wrike also contains Content-Type and Contenttype

**Comment by mikegioia** [comment]
This is happened for headers other than Content-Type, FYI. I ran into this problem with Mime-Version coming in as MimeVersion. I implemented it in my fork, here's an example: https://github.com/ParticleBits/zend-mail/commit/96a8a4a3b0803a05ba1665e87e5a3a4a255ffcff

I don't know if this would be the preferred fix for this codebase but if it is, it's a minor PR.

**Comment by michalbundyra** [comment]
This repository has been closed and moved to laminas/laminas-mail; a new issue has been opened at https://github.com/laminas/laminas-mail/issues/51.
